python/pyspark/sql/context.py

```diff
@@ -202,6 +202,32 @@ def registerFunction(self, name, f, returnType=StringType()):
         """
         self.sparkSession.catalog.registerFunction(name, f, returnType)
 
+    @ignore_unicode_prefix
+    @since(2.1)
+    def registerJavaFunction(self, name, javaClassName, returnType=None):
+        """Register a java UDF so it can be used in SQL statements.
+
+        In addition to a name and the function itself, the return type can be optionally specified.
+        When the return type is not given it would infer the returnType via reflection.
```

Comment:
nit: its a little odd to mix `return type` with `returnType`.  Perhaps, "When the return type is not specified we attempt to infer it using reflection"